### DISCUSSION TITLE
Popup overlays text input

### DISCUSSION THREAD

**Discussion by marcuspuchalla** [minad/corfu#494] [Q&A] [answered]
I'm using corfu with emacs 29.3-3 nativecomp on arch with exwm.

My problem is that the corfu overlay is alway in the line where I type. So as soon as the popup is show, I can not see what I am typing.

Any ideas on what could cause this ? Maybe it's exwm related but I could not find any info. Any help would be appreciated.

![corfu_hides_input_2024-08-18_10-36](https://github.com/user-attachments/assets/18d274e1-8d80-4c5e-8507-22efd0e23eaa)

This is part of my current emacs init.
```

(defun orderless-fast-dispatch (word index total)
  (and (= index 0) (= total 1) (length< word 4)
       (cons 'orderless-literal-prefix word)))

(orderless-define-completion-style orderless-fast
  (orderless-style-dispatchers '(orderless-fast-dispatch))
  (orderless-matching-styles '(orderless-literal orderless-regexp)))

     ;;; Code:
(use-package corfu
  :ensure t
  ;; Optional customizations
  :custom
  (corfu-cycle t)                 ; Allows cycling through candidates
  (corfu-auto t)                  ; Enable auto completion
  (corfu-auto-prefix 2)           ; Minimum length of prefix for completion
  (corfu-separator ?\s)          ;; Orderless field separator

  (corfu-auto-delay 0.5)            ; No delay for completion
  (corfu-popupinfo-delay '(0.5 . 0.2))  ; Automatically update info popup after that numver of seconds
  (corfu-preview-current 'insert) ; insert previewed candidate
  (corfu-preselect 'prompt)
  (corfu-on-exact-match nil)      ; Don't auto expand tempel snippets
  (corfu-auto t)
  (corfu-quit-no-match 'separator)
  (completion-styles '(orderless-fast basic))

  ;; Optionally use TAB for cycling, default is `corfu-complete'.
  :bind (:map corfu-map
	      ("M-SPC"      . corfu-insert-separator)
	      ("TAB"        . corfu-next)
	      ([tab]        . corfu-next)
	      ("S-TAB"      . corfu-previous)
	      ([backtab]    . corfu-previous)
	      ("S-<return>" . corfu-insert)
	      ("RET"        . corfu-insert))

  :init
  (global-corfu-mode)
  (corfu-history-mode)
  (corfu-popupinfo-mode) ; Popup completion info
  :config
  (add-hook 'eshell-mode-hook
	    (lambda () (setq-local corfu-quit-at-boundary t
				   corfu-quit-no-match t
				   corfu-auto nil)
	      (corfu-mode))
	    nil
	    t))

;; A few more useful configurations...
(use-package emacs
  :custom
  ;; TAB cycle if there are only few candidates
  ;; (completion-cycle-threshold 3)

  ;; Enable indentation+completion using the TAB key.
  ;; `completion-at-point' is often bound to M-TAB.
  (tab-always-indent 'complete)

  ;; Emacs 30 and newer: Disable Ispell completion function. As an alternative,
  ;; try `cape-dict'.
  ;; @ToDo: enable once upgraded to emacs 30
  ;; (text-mode-ispell-word-completion nil)

  ;; Emacs 28 and newer: Hide commands in M-x which do not apply to the current
  ;; mode.  Corfu commands are hidden, since they are not used via M-x. This
  ;; setting is useful beyond Corfu.
  (read-extended-command-predicate #'command-completion-default-include-p))

```

**Comment by minad** [comment]
Hi! Thanks for the report. I also use Corfu on EXWM without issues. I've never seen such an issue, where the popup position is computed incorrectly. Do you use a single monitor? Do you use an additional compositor? Do you use special font configurations? Can you try to reproduce the problem with a minimal configuration based on emacs -Q, such that we get minimal interference from other subsystems?

**Reply by marcuspuchalla** [reply]
Hi there, thanks for quick response which already guides me into the correct direction.
For your questions, I'm on a single monitor with no additional compositor. I do not have special fonts, but a custom theme which might cause the problem. But not sure. 

Nevertheless I did create a minimal config and in there, I do not have the problem. So I guess it is some other configuarion in my emacs init file. I will now deep dive and check which setting is breaking the lines.

I will report back here what I found, maybe it will help other.

**Comment by marcuspuchalla** [comment]
The problem comes from my ploybar it seems. When this is started, I get this error.
I tried and moved starting the ploybar from my emacs config to the script that is starting my emacs client, but no success. Still the same problem. :thinking:

**Reply by marcuspuchalla** [reply]
This seems very related https://github.com/emacs-exwm/exwm/issues/31

**Comment by marcuspuchalla** [comment, marked as answer]
In fact, the code [here](https://github.com/emacs-exwm/exwm/issues/31#issuecomment-2029704237) seems to solve the problem.
